ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/txselection/BlockTransactionSelector.java

```diff
@@ -422,14 +466,16 @@ private TransactionSelectionResult handleTransactionNotSelected(
     final var pendingTransaction = evaluationContext.getPendingTransaction();
 
     // check if this tx took too much to evaluate, and in case it was invalid remove it from the
-    // pool, otherwise penalize it.
+    // pool, otherwise penalize it. Not synchronized since there is no state change here.
     final TransactionSelectionResult actualResult =
```

Comment:
This becomes quite complex to read. Can we put this into a method with a clear name?

Reply:
yes will rework to make it easier to read

Reply:
I have rework it a bit, and added javadoc, hope it is clearer now